assembly/src/ast/mod.rs

```diff
@@ -410,25 +438,31 @@ impl Serializable for ModuleAst {
 
         match &self.docs {
             Some(docs) => {
-                assert!(docs.len() <= u16::MAX as usize, "docs too long");
+                assert!(docs.len() <= MAX_DOCS_LEN, "docs too long");
                 target.write_u16(docs.len() as u16);
                 target.write_bytes(docs.as_bytes());
             }
             None => {
                 target.write_u16(0);
             }
         }
-
-        assert!(self.local_procs.len() <= u16::MAX as usize, "too many local procs");
-        target.write_u16(self.local_procs.len() as u16);
+        assert!(self.imports.len() <= MAX_IMPORTS, "too many imports");
+        assert!(self.local_procs.len() <= MAX_LOCAL_PROCS, "too many local procs");
+        assert!(
+            self.reexported_procs.len() <= MAX_REEXPORTED_PROCS,
+            "too many re-exported procs"
+        );
+        target.write_u16(self.imports.len() as u16);
+        self.imports.values().for_each(|i| i.write_into(target));
```

Comment:
Based on @bobbinth's comment above (https://github.com/0xPolygonMiden/miden-vm/pull/940#discussion_r1239336708), we should probably still keep it optional. I've changed the serialization and deserialization of libraries to serialize and deserialize dependencies independently and not depend on `ModuleAst`. This would mean though that there is some redundancy in case we do serialize imports in `ModuleAst`.